src/confluent_kafka/src/confluent_kafka.c

```diff
@@ -783,33 +894,85 @@ static int Message_traverse(Message *self, visitproc visit, void *arg) {
         return 0;
 }
 
+static PyObject *Message_richcompare(PyObject *self, PyObject *other, int op) {
+        if (op != Py_EQ && op != Py_NE) {
+                Py_INCREF(Py_NotImplemented);
+                return Py_NotImplemented;
+        }
+
+        if (self == other) {
+                return op == Py_EQ ? Py_True : Py_False;
+        }
+
+        if (!PyObject_TypeCheck(other, &MessageType)) {
+                return op == Py_EQ ? Py_False : Py_True;
+        }
+
+        Message *msg_self  = (Message *)self;
+        Message *msg_other = (Message *)other;
+
+        int result;
+
+#define _LOCAL_COMPARE(left, right)                                            \
+        do {                                                                   \
+                result = PyObject_RichCompareBool(left, right, Py_EQ);         \
+                if (result < 0)                                                \
+                        return NULL;                                           \
+                if (result == 0)                                               \
+                        return op == Py_EQ ? Py_False : Py_True;               \
+        } while (0)
```

Comment:
Forgot about the macro define while(0) trick to make single line statement blocks in C.